### PR TITLE
DAOS-14203 test: Test memory usage after server start/stop

### PR DESCRIPTION
**Pull request by phender**
Skip-unit-tests: true
Skip-fault-injection-test: true
Allow-unstable-test: true
Test-tag: test_server_memory_usage

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/control/daos_server_mem_usage.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            dict: total, free, and available memory per server host
+        """
+        memory_data = {}
+        result = run_remote(self.log, self.server_managers[0].hosts, "free -m", stderr=True)
+        if not result.passed:
+            self.fail(f'Error obtaining available memory on {result.failed_hosts}')
+        for data in result.output:
+            hosts = str(data.hosts)
+            try:
+                info = re.findall('Mem:\s+(\d+)\s+\d+\s+(\d+)\s+\d+\s+\d+\s+(\d+)', data.stdout)
```
Comment (daltonbohning) [inline src/tests/ftest/control/daos_server_mem_usage.py:29]: flake wants you to do this
```suggestion
                info = re.findall(r'Mem:\s+(\d+)\s+\d+\s+(\d+)\s+\d+\s+\d+\s+(\d+)', data.stdout)
```

Comment (phender) [reply src/tests/ftest/control/daos_server_mem_usage.py]: Fixed.

**Review by daltonbohning — DISMISSED** [review]

**Review by tanabarr — COMMENTED** [review]
can you see if the hugepages are contributing to the memory leak?

File: src/tests/ftest/control/daos_server_mem_usage.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  start_servers_once: False
+server_config:
+  name: daos_server
+  port: 10001
+  engines_per_host: 1
+  engines:
+    0:
+      storage: auto
+sequences: 3
+memory_restored_min_percent: 95
```
[the remark this answers is not in the document]
Comment (phender) [reply src/tests/ftest/control/daos_server_mem_usage.yaml:17]: Just a ballpark figure initially.  I'd welcome some input if we have a better expectation.

**Comment by phender** [comment]
> can you see if the hugepages are contributing to the memory leak?

I've added the suggested `cat /proc/meminfo | grep Huge` command to the debug info we display when starting and now stopping the servers.  We'll see if it highlights anything in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13165/7/.

**Review by daltonbohning — APPROVED** [review]

File: src/tests/ftest/control/daos_server_mem_usage.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tanabarr) [reply src/tests/ftest/control/daos_server_mem_usage.yaml:17]: I imagine anything above 90% is probably okay

**Review by tanabarr — APPROVED** [review]

**Comment by phender** [comment]
Ticket https://daosio.atlassian.net/browse/DAOS-14203 has been resolved.

**Pull request closed without merging**